dask/array/reblock.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import, division, print_function
+"""
+The reblock module defines:
+    intersect_blockdims: a function for
+        converting blockdims to new dimensions
+    reblock: a function to convert the blocks
+        of an existing dask array to new blockdims or blockshape
+"""
+
+from itertools import count, product, chain
+from operator import getitem, add
+import numpy as np
+from toolz import merge, accumulate
+from dask.array.core import rec_concatenate, Array
+from dask.array.core import blockdims_from_blockshape
+
+
+reblock_names  = ('reblock-%d' % i for i in count(1))
+
+
+def cumdims_label(blockdims, const):
+    """ Interal utility for cumulative sum with label.
+
+    >>> cumdims_label(((5, 3, 3), (2, 2, 1)), 'n')  # doctest: +NORMALIZE_WHITESPACE
+    [(('n', 0), ('n', 5), ('n', 8), ('n', 11)),
+     (('n', 0), ('n', 2), ('n', 4), ('n', 5))]
+    """
+    return [tuple(zip((const,) * (1 + len(bds)),
+                      list(accumulate(add, (0,) + bds))))
+              for bds in blockdims ]
+
+
+def _breakpoints(cumold, cumnew):
+    """
+
+    >>> new = cumdims_label(((5, 3, 3), (2, 2, 1)), 'n')
+    >>> old = cumdims_label(((2, 2, 1), (5,)), 'o')
```

Comment:
I think it would be easier if you could fix it in yours, just to make sure
I don't commit any other partial changes I have made yesterday/today.
Thank you,

On Mon, Mar 30, 2015 at 10:15 AM, Matthew Rocklin <notifications@github.com>
wrote:

> In dask/array/reblock.py
> <https://github.com/ContinuumIO/dask/pull/113#discussion_r27410772>:
>
> > +    """ Interal utility for cumulative sum with label.
> > +
> > +    >>> cumdims_label(((5, 3, 3), (2, 2, 1)), 'n')  # doctest: +NORMALIZE_WHITESPACE
> > +    [(('n', 0), ('n', 5), ('n', 8), ('n', 11)),
> > +     (('n', 0), ('n', 2), ('n', 4), ('n', 5))]
> > +    """
> > +    return [tuple(zip((const,) * (1 + len(bds)),
> > +                      list(accumulate(add, (0,) + bds))))
> > +              for bds in blockdims ]
> > +
> > +
> > +def _breakpoints(cumold, cumnew):
> > +    """
> > +
> > +    >>> new = cumdims_label(((5, 3, 3), (2, 2, 1)), 'n')
> > +    >>> old = cumdims_label(((2, 2, 1), (5,)), 'o')
>
> If pulling down my branch, making a new commit, and then issuing a PR to
> my branch is a bit of a pain then I'm also happy to implement the fix. Just
> curious what it should be.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/113/files#r27410772>.
>



-- 
Peter Steinberg - Continuum Analytics
206 715 4492